106-auto-rebalance-on-imbalanced-clusters.md

```diff
@@ -0,0 +1,524 @@
+This field is optional and if not specified, the auto-rebalancing runs with the default Cruise Control configuration (i.e. the same used for unmodified manual `KafkaRebalance` invocations).
+To provide users more flexibility, they only have to configure the auto-rebalance modes they wish to use whether it be `add-brokers`, `remove-brokers`, or `imbalance`..
+When the auto-rebalance configuration is set with `imbalance` mode enabled, the operator will trigger a partition rebalance whenever a goal violation is detected by the anomaly detector.
+To trigger the auto-rebalance, the operator must know that the cluster is imbalanced due to some goal violation anomaly. 
+We will create our own custom notifier named `StrimziCruiseControlNotifier` to do the same.
+This notifier's job will be to update the operator regarding the goal violations so that the operator can trigger a rebalance (see section [AnomalyDetectorNotifier](./106-auto-rebalance-on-imbalanced-clusters.md#anomalydetectornotifier)).
+With this proposal, we are only going to support auto-rebalance on imbalanced cluster.
+We also plan to implement the same for topic and metrics related issues, but it will be part of future work since their implementation require different approach.
+For example, when dealing with topic related issues, it will require a coordination with topic operator and metrics issues will require coordination with the Kafka API.
+
+Here is an example of what the configured `Kafka` custom resource could look like:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+  # ...
+  cruiseControl:
+    # ...
+    autoRebalance:
+      # using the imbalance rebalance mode
+      - mode: imbalance
+        template:
+          name: my-imbalance-rebalance-template
+```
+
+It is also possible to use the default Cruise Control rebalancing configuration by omitting the `template` field.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+  cruiseControl:
+    # ...
+    autoRebalance:
+      # using the default Cruise Control rebalancing configuration 
+      - mode: imbalance
+```
+
+The auto-rebalance configuration for the `spec.cruiseControl.autoRebalance.template` property in the `Kafka` custom resource is provided through a `KafkaRebalance` custom resource defined as a "template".
+That is a `KafkaRebalance` custom resource with the `strimzi.io/rebalance-template: true` annotation set.
+When it is created, the `KafkaRebalanceAssemblyOperator` doesn't run any rebalancing.
```

Comment:
I would just mention the Strimzi Cluster Operator than the specific assembly operator class.